protocol/chainlib/rest.go

```diff
@@ -240,7 +242,9 @@ func (apil *RestChainListener) Serve(ctx context.Context) {
 		analytics := metrics.NewRelayAnalytics(dappID, chainID, apiInterface)
 
 		reply, _, err := apil.relaySender.SendRelay(ctx, path, query, http.MethodGet, dappID, analytics)
-		go apil.logger.AddMetric(analytics, err != nil)
+		if apil.logger.ShouldCountMetricForHttp(c) {
```

Comment:
make a wrapper method to wrap this check and runn addmetric inside, this method should run in a go routine instead the add metric one. so we do the check in a routine as well.

Reply:
i put everything inside add metric will push the changes now